### PR TITLE
Implements #470: Verifies that box width is valid

### PR DESCRIPTION
**Pull request by badisa**
* If cutoff * 2 is greater than the min box width then
  the neighborlist construction is poorly defined since
  a particle can overlap with another tile more than once

### CODE REVIEW

**Comment by badisa** [comment]
Based on the slow_tests/test_benchmarks.py, this has no impact upon performance.

File: timemachine/cpp/src/nonbonded.cu [hunk 1]
```diff
@@ -303,6 +303,23 @@ void Nonbonded<RealType, Interpolated>::execute_device(
...
+        // Verify that the cutoff and box size are valid together. If cutoff is greater than half the box
+        // then a particle can interact with multiple periodic copies.
+        std::vector<double> h_box(9);
+        gpuErrchk(cudaMemcpyAsync(&h_box[0], d_box, 3*3*sizeof(*d_box), cudaMemcpyDeviceToHost, stream));
+        const double twice_sq_cutoff = 2 * ((cutoff_+nblist_padding_) * (cutoff_+nblist_padding_));
+
+        cudaStreamSynchronize(stream);
+        // Use l2 norm on each row to determine if cutoff is too small
+        for (int i = 0; i < 3; i++) {
+            double row_norm_sq = h_box[i*3+0]*h_box[i*3+0] + h_box[i*3+1]*h_box[i*3+1] + h_box[i*3+2]*h_box[i*3+2];
```
Comment (proteneer) [inline timemachine/cpp/src/nonbonded.cu:316]: looks right to me, note that we should document somewhere that the box 3x3 matrix corresponds to:
```
[
   [u_x,u_y,u_z],
   [v_x,v_y,v_z], 
   [w_x,w_y,w_z],
]
```

where `u,v,w` correspond to be the box vectors. Different packages have different conventions.

File: tests/test_parameter_interpolation.py [hunk 2]
```diff
@@ -102,9 +100,7 @@ def test_nonbonded_advanced(self):
         water_coords = self.get_water_coords(D, sort=False)
         coords = water_coords[:size]
         padding = 0.2
-        diag = np.amax(coords, axis=0) - np.amin(coords, axis=0) + padding
-        box = np.eye(3)
-        np.fill_diagonal(box, diag)
+        box = np.eye(3) * 3
```
Comment (proteneer) [inline tests/test_parameter_interpolation.py:104]: add a test for a non-orthogonal box just to sanity check that the test would fail?

**Comment by badisa** [comment]
Closing as #498 made a rebase messy

**Pull request closed without merging**